macos/Classes/FilePickerPlugin.swift

```diff
@@ -0,0 +1,123 @@
+import Cocoa
+import FlutterMacOS
+
+public class FilePickerPlugin: NSObject, FlutterPlugin {
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    let channel = FlutterMethodChannel(
+      name: "miguelruivo.flutter.plugins.filepicker",
+      binaryMessenger: registrar.messenger)
+    let instance = FilePickerPlugin()
+    registrar.addMethodCallDelegate(instance, channel: channel)
+  }
+
+  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult)
+  {
+    switch call.method {
+    case "pickFiles":
+      handleFileSelection(call, result: result)
+
+    case "getDirectoryPath":
+      handleDirectorySelection(call, result: result)
+
+    case "saveFile":
+      handleSaveFile(call, result: result)
+
+    default:
+      result(FlutterMethodNotImplemented)
+    }
+  }
+
+  private func handleFileSelection(
+    _ call: FlutterMethodCall, result: @escaping FlutterResult
+  ) {
+    let dialog = NSOpenPanel()
+    let args = call.arguments as! [String: Any]
+
+    dialog.showsResizeIndicator = true
+    dialog.directoryURL = URL(
+      fileURLWithPath: args["initialDirectory"] as? String ?? ""
+    )
+    dialog.showsHiddenFiles = false
+    dialog.allowsMultipleSelection = args["allowMultiple"] as? Bool ?? false
+    dialog.canChooseDirectories = false
+    dialog.allowedFileTypes = args["allowedExtensions"] as? [String] ?? []
+
+    if dialog.runModal() == NSApplication.ModalResponse.OK {
+      let pathResult = dialog.url
```

Comment:
This doesn't look like it is correct? The `url` is probably nil when in multiple selection mode.
Can you check `dialog.urls` directly instead? That is an array so you'd need the first item when in single selection mode (or return nil if it's empty)

Reply:
For multiple selections the `.url` includes the first of the files, so it's safe to access it for check, but I think you are right that to reduce confusion it may be better to check `.urls`

![screenshot_20250129_134502](https://github.com/user-attachments/assets/587018ba-9605-412e-bd29-1909d5b0acef)